client/signup/steps/p2-complete-profile/index.jsx

```diff
@@ -0,0 +1,260 @@
+			form: this.state.form,
+		} );
+	};
+
+	handleBlur = () => {
+		this.formStateController.sanitize();
+		this.save();
+	};
+
+	handleChangeEvent = ( event ) => {
+		this.formStateController.handleFieldChange( {
+			name: event.target.name,
+			value: event.target.value,
+		} );
+	};
+
+	getErrorMessages = ( fieldName ) => {
+		const messages = formState.getFieldErrorMessages( this.state.form, fieldName );
+
+		if ( ! messages ) {
+			return;
+		}
+
+		return messages;
+	};
+
+	formFields = () => {
+		const fieldDisabled = this.state.isSubmitting;
+
+		return (
+			<>
+				<ValidationFieldset
+					errorMessages={ this.getErrorMessages( 'fullName' ) }
+					className="p2-complete-profile__validation-full-name"
+				>
+					<FormLabel htmlFor="full-name-input">
+						{ this.props.translate( 'Your Full Name' ) }
+					</FormLabel>
+					<FormTextInput
+						id="full-name-input"
+						autoFocus={ true } // eslint-disable-line jsx-a11y/no-autofocus
+						autoCapitalize={ 'off' }
+						className="p2-complete-profile__full-name"
+						disabled={ fieldDisabled }
+						name="full-name"
+						value={ formState.getFieldValue( this.state.form, 'fullName' ) }
+						isError={ formState.isFieldInvalid( this.state.form, 'fullName' ) }
+						isValid={ formState.isFieldValid( this.state.form, 'fullName' ) }
+						onBlur={ this.handleBlur }
+						onChange={ this.handleChangeEvent }
+					/>
+				</ValidationFieldset>
+			</>
+		);
+	};
+
+	renderUploadAvatarBtn = () => {
+		return (
+			<button className="p2-complete-profile__upload-avatar-btn">
+				{ this.props.translate( 'Upload a new avatar' ) }
+			</button>
+		);
+	};
+
+	render() {
+		return (
+			<P2StepWrapper
+				flowName={ this.props.flowName }
+				stepName={ this.props.stepName }
+				positionInFlow={ this.props.positionInFlow }
+				headerText={ this.props.translate( 'Complete your profile' ) }
+				subHeaderText={ this.props.translate(
+					'Using a recognizable photo and name will help your team to identify you more easily.'
+				) }
+			>
+				<div className="p2-complete-profile">
+					<div className="p2-complete-profile__avatar-wrapper">
+						<EditGravatar additionalUploadHtml={ this.renderUploadAvatarBtn() } />
+					</div>
+
+					<div className="p2-complete-profile__form-wrapper">
+						<form className="p2-complete-profile__form" onSubmit={ this.handleSubmit } noValidate>
+							{ this.formFields() }
+							<div className="p2-complete-profile__form-footer">
+								<FormButton
+									disabled={ this.state.isSubmitting }
+									className="p2-complete-profile__form-submit-btn"
+								>
+									{ this.props.translate( 'Continue' ) }
+								</FormButton>
+							</div>
+						</form>
+					</div>
+
+					<div className="p2-complete-profile__skip-wrapper">
+						<span>
+							No time? No problem! You can{ ' ' }
+							<Button
+								className="p2-complete-profile__skip-btn"
+								variant="link"
+								onClick={ () => {
```

Comment:
Can we abstract that callback into a method so it doesn't get redeclared on every re-render?